File: src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/SerializedChatInteraction.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class SerializedChatInteraction(
  val humanMessage: SerializedChatMessage,
  val assistantMessage: SerializedChatMessage? = null,
)

File: agent/bindings/csharp/SerializedChatInteraction.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class SerializedChatInteraction
  {
    [JsonProperty(PropertyName = "humanMessage")]
    public SerializedChatMessage HumanMessage { get; set; }
    [JsonProperty(PropertyName = "assistantMessage")]
    public SerializedChatMessage AssistantMessage { get; set; }
  }
}
